Informatics II/Lectures/Twihning/heapsort.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <math.h>
+#define N 13
+
+int arr[N] = {0,2,7,8,3,9,4,1,6,11,14,12,17};
+
+//function to swap 2 elements
+void swap(int *a, int i, int j){
+        int temp = a[i];
+        a[i] = a[j];
+        a[j] = temp;
+}
+//determine the index of right child 
+int right(int i){
+    return i*2+2;
+}
+
+//determine the index of left child 
+int left(int i){
+    return i*2+1;
+}
+
+//convert the tree to a heap from the index i downwards
+void heapify(int *A, int i , int k){
+    int m, l, r;
+    m = i;
+    l = left(i);
+    r = right(i);
```

Comment:
This can be shortened as:
```suggestion
    int l = left(i);
    int r = right(i);
```